app/src/main/java/com/hedvig/app/feature/profile/ui/payment/PaymentHistoryAdapter.kt

```diff
@@ -10,21 +10,27 @@ import kotlinx.android.synthetic.main.payment_history_item.view.*
 import kotlinx.android.synthetic.main.payout_history_header.view.*
 import kotlinx.android.synthetic.main.payout_history_title.view.*
 
-class PaymentHistoryAdapter(
-    private val items: List<ChargeWrapper>
-) : RecyclerView.Adapter<PaymentHistoryAdapter.ViewHolder>() {
+class PaymentHistoryAdapter : RecyclerView.Adapter<PaymentHistoryAdapter.ViewHolder>() {
+
+    var items: List<ChargeWrapper> = listOf()
+
     override fun getItemCount() = items.size
 
-    override fun getItemViewType(position: Int) = when (items[position]) {
-        is ChargeWrapper.Header -> HEADER
-        is ChargeWrapper.Item -> ITEM
-        is ChargeWrapper.Title -> TITLE
+    override fun getItemViewType(position: Int) = when (position) {
+        0 -> TITLE
+        1 -> HEADER
+        else -> ITEM
     }
 
     override fun onCreateViewHolder(
         parent: ViewGroup,
         viewType: Int
     ): ViewHolder = when (viewType) {
+        TITLE -> ViewHolder.TitleViewHolder(
```

Comment:
Here we can refactor the `ViewHolder` to have the responsibility of inflating its layout in the constructor, similar to how we do in other more recent `RecyclerView.Adapter`s 🙂

Reply:
Also in the `onBindViewHolder`-method, we can refactor the `bind`-functionality to be a part of the `ViewHolder` as well, giving the `ViewHolder` complete ownership of both its layout and its data binding procedure.